pilot/pkg/model/telemetry.go

```diff
@@ -1086,6 +1096,40 @@ var metricToPrometheusMetric = map[string]string{
 	"GRPC_RESPONSE_MESSAGES": "response_messages_total",
 }
 
+// metricRotationInterval returns the metric rotation interval for the proxy,
+// only work fro istio's metrics. if nil is returned, it means disable rotation.
+func metricRotationInterval(proxy *Proxy) *durationpb.Duration {
+	// get the metric rotation interval from the proxy metadata if it exists
+	return parseDurationFromProxyMetadata(proxy, "ISTIO_META_METRIC_ROTATION_INTERVAL", 0*time.Second)
+}
```

Comment:
A nit, I've seen this in other places: ISTIO_META prefix was intended for env variables in agent which get added to the bootstrap, with the prefix removed.  I suspect at a later point someone decided to add ALL labels and annotations, including ProxyConfig ? As a result our bootstrap is insanely large and it became impossible to figure out what is actually needed...

For example ISTIO_META_DNS_CAPTURE becomes DNS_CAPTURE - which is declared directly in NodeMetadata. 

I think using proxy.Metadata.ProxyConfig.ProxyMetadata["ISTIO_META_FOO"] instead of 
proxy.Metadata.Foo is not ideal - also passing back ProxyConfig is not ideal.

Reply:
@howardjohn maybe a larger topic - but the UX is also pretty bad, ProxyConfig annotation is a good idea from code perspective but it becomes very hard to use. 

I am going to try to push back on new env variables and global settings - it is unsafe to do global experiments - but that will push more stuff into ProxyConfig.ProxyMetadata ( and I am not clear what we want to do with the ProxyConfig CRD, or if it is merged or read in the path - can a user set this telemetry metadata in the CRD and have a dynamic update ? Would be very nice ...)

Reply:
Please make sure we make these decisions project-wide and not on individual PRs. Whether env var vs proxy config vs annotation is better we can debate. But our choice should not be "depends on which PRs Costin noticed and suggested X" but rather "we have a global policy to do X". Its important to be consistent